src/validator/CreditCardValidator.java

```diff
@@ -0,0 +1,62 @@
+package validator;
+
+import java.util.ArrayList;
+
+public class CreditCardValidator implements CreditCardChecking {
+
+	@Override
+	public boolean checkValidity(CreditCardVendor ccv, String number) {
+		ArrayList<Integer> chars = ccv.getNumberOfChars();
+		ArrayList<String> masks = ccv.getIIN();
+		
+		boolean result1 = checkLengthCorrectness(number, chars);
+		boolean result2 = checkIINMaskCorrectness (number, masks);
+		boolean result3 = checkLuhnalgorithmCorrectness(number);
+		
+		return result1 && result2 && result3;
+	}
+	
+	private boolean checkLengthCorrectness(String number, ArrayList<Integer> chars){
+		boolean result = false;
+		for (int i = 0; i<chars.size();i++){
+			if (chars.get(i) == number.length())
+				result = true;
+		}
+		return result;
+	}
+	
+	private boolean checkIINMaskCorrectness (String number, ArrayList<String> masks){
+		boolean result = false;
+		for (int i = 0; i<masks.size();i++){
+			if (number.startsWith(masks.get(i)))
+				result = true;
+		}
+		return result;
+	}
+	
+	private boolean checkLuhnalgorithmCorrectness(String number){
```

Comment:
algorithm should begin with a capital letter. Should be: checkLuhnAlgorithmCorrectness

Reply:
It is worth considering whether to use an external implementation of the algorithm. E.g. 
Class LuhnCheckDigit from org.apache.commons.validator.routines.checkdigit
https://commons.apache.org/proper/commons-validator/apidocs/org/apache/commons/validator/routines/checkdigit/LuhnCheckDigit.html
Depending on whether you want to depend on external solutions